Tools/Cli-LoRa-Device-Provisioning/ValidationHelper.cs

```diff
@@ -0,0 +1,445 @@
+        {
+            var isValid = true;
+            var isWarning = false;
+
+            if (sensorDecoder == null)
+            {
+                StatusConsole.WriteLine(MessageType.Error, "SensorDecoder is missing.");
+                return false;
+            }
+
+            if (sensorDecoder == string.Empty)
+            {
+                StatusConsole.WriteLine(MessageType.Info, "SensorDecoder is empty. No decoder will be used.");
+                return true;
+            }
+
+            if (sensorDecoder.StartsWith("http") || sensorDecoder.Contains('/'))
+            {
+                if (!Uri.TryCreate(sensorDecoder, UriKind.Absolute, out Uri validatedUri))
+                {
+                    StatusConsole.WriteLine(MessageType.Error, "SensorDecoder has invalid URL.");
+                    isValid = false;
+                }
+
+                // if (validatedUri.Host.Any(char.IsUpper))
+                if (!sensorDecoder.Contains(validatedUri.Host))
+                {
+                    StatusConsole.WriteLine(MessageType.Error, "SensorDecoder Hostname must be all lowercase.");
+                    isValid = false;
+                }
+
+                if (validatedUri.AbsolutePath.IndexOf("/api/") < 0)
+                {
+                    StatusConsole.WriteLine(MessageType.Warning, "SensorDecoder is missing \"api\" keyword.");
+                    isWarning = true;
+                }
+            }
+
+            if (!isValid || isWarning)
+            {
+                StatusConsole.WriteLine(MessageType.Info, "Make sure the URI based SensorDecoder Twin desired property looks like \"http://containername/api/decodername\".");
+            }
+            else
+            {
+                StatusConsole.WriteLine(MessageType.Info, $"SensorDecoder {sensorDecoder} is valid.");
+            }
+
+            return isValid;
+        }
+
+        public static bool ValidateFcntSettings(Program.AddOptions opts, string fCntUpStartReported, string fCntDownStartReported, string fCntResetCounterReported)
+        {
+            var isValid = true;
+
+            var abpRelaxMode = string.IsNullOrEmpty(opts.ABPRelaxMode) ||
+                string.Equals(opts.ABPRelaxMode, "1", StringComparison.InvariantCultureIgnoreCase) ||
+                string.Equals(opts.ABPRelaxMode, "true", StringComparison.InvariantCultureIgnoreCase);
+
+            var fCntUpStart = SetUIntProperty(opts.FCntUpStart);
+            var fCntDownStart = SetUIntProperty(opts.FCntDownStart);
+            var fCntResetCounter = SetUIntProperty(opts.FCntResetCounter);
+
+            var fCntUpStartRep = SetUIntProperty(fCntUpStartReported);
+            var fCntDownStartRep = SetUIntProperty(fCntDownStartReported);
+            var fCntResetCounterRep = SetUIntProperty(fCntResetCounterReported);
+
+            // AbpRelaxMode TRUE
+            if (abpRelaxMode)
+            {
+                if (fCntUpStart != null)
```

Comment:
@Mandur : can you help shed light here?

Reply:
Yeah I think it is correct. To my understanding, relax frame counter enable any value > current fcnt, so it could be anything. Note this is not something from the specification therefore I'd be happy if @ronniesa  validate.

Reply:
Changed based on Patricks's input.